src/wallet/file.rs

```diff
@@ -21,35 +19,71 @@ pub trait SecureWalletFile {
 
 /// Wrapper around `PathBuf` for wallet paths
 #[derive(PartialEq, Eq, Hash, Debug, Clone)]
-pub struct WalletPath(pub(crate) PathBuf);
+pub struct WalletPath {
+    /// Path of the wallet file
+    pub wallet: PathBuf,
+    /// Directory of the cache
+    pub cache: PathBuf,
+    /// Name of the network
+    pub network: Option<String>,
+}
 
 impl WalletPath {
     /// Create a new wallet path from a directory and a name
     pub fn new(dir: &Path) -> Self {
         let pb = PathBuf::from(dir);
-        Self(pb)
+        // Usually wallet path is .dusk-dir/wallet/wallet.dat, by default
+        // use this dir, else specify one with set_cache_dir
+        let mut cache = pb.clone();
+
+        cache.pop();
+
+        Self {
+            wallet: pb,
+            cache,
+            network: None,
+        }
     }
 
     /// Returns the filename of this path
     pub fn name(&self) -> Option<String> {
         // extract the name
-        let name = self.0.file_stem()?.to_str()?;
+        let name = self.wallet.file_stem()?.to_str()?;
         Some(String::from(name))
     }
 
     /// Returns current directory for this path
     pub fn dir(&self) -> Option<PathBuf> {
-        self.0.parent().map(PathBuf::from)
+        self.wallet.parent().map(PathBuf::from)
     }
 
     /// Returns a reference to the `PathBuf` holding the path
     pub fn inner(&self) -> &PathBuf {
-        &self.0
+        &self.wallet
     }
 
     /// Sets the directory for the state cache
-    pub fn set_cache_dir(path: &Path) -> Result<(), Error> {
-        State::set_cache_dir(path.to_path_buf())
+    pub fn set_cache_dir(&mut self, path: &Path) {
+        self.cache = path.to_path_buf();
+    }
+
+    /// Sets the network name for different cache locations.
+    /// e.g, devnet, testnet, etc.
+    pub fn set_network_name(&mut self, network: Option<String>) {
+        self.network = network;
+    }
+
+    /// Generates dir for cache based on network specified
+    pub fn cache_dir(&self) -> PathBuf {
+        let mut cache = self.cache.clone();
+
+        if let Some(network) = &self.network {
+            cache.push(network);
```

Comment:
Add a prefix to the network (eg: `cache_`) so to not collide with existing files